op-node/rollup/driver/finalized.go

```diff
@@ -0,0 +1,22 @@
+package driver
+
+import (
+	"context"
+
+	"github.com/ethereum-optimism/optimism/op-node/rollup/derive"
+	"github.com/ethereum-optimism/optimism/op-service/eth"
+)
+
+type finalized struct {
+	derive.L1Fetcher
+}
+
+func NewFinalized(fetcher derive.L1Fetcher) *finalized {
+	return &finalized{L1Fetcher: fetcher}
+}
+
+func (f *finalized) L1BlockRefByNumber(ctx context.Context, num uint64) (eth.L1BlockRef, error) {
+	return f.L1BlockRefByLabel(ctx, eth.Finalized)
```

Comment:
Also, this is quite incorrect!

If i ask for block number XXX, you can't JUST resturn the last finalized block! Instead you should return the block XXX if and only if it's already finalized

Reply:
Good point, not sure what I was thinking. Fixed in the latest commit.